src/diffusers/models/attention_processor.py

```diff
@@ -3938,7 +3950,7 @@ def apply_rotary_emb(x, freqs_cos, freqs_sin):
         # dropout
         hidden_states = attn.to_out[1](hidden_states)
 
-        if hasattr(attn, "to_add_out"):
+        if attn.context_pre_only is not None and not attn.context_pre_only:
```

Comment:
`getattr(attn, "to_add_out", None) is not None` is better because it's more direct and easy to understand, no?
i.e. we only run this layer when we have this layer

Reply:
Okay yes, that sounds easier to understand. Will update